client/components/editor/teaching-elements/Assessment/ImageMultipleChoice.vue

```diff
@@ -0,0 +1,326 @@
+      </li>
+    </ul>
+  </div>
+</template>
+
+<script>
+import cloneDeep from 'lodash/cloneDeep';
+import range from 'lodash/range';
+import { blobToDataURL } from 'blob-util';
+
+const maxImageDimension = 256;
+
+const formAlert = (message) => {
+  return {
+    type: 'alert-danger',
+    text: message
+  };
+};
+
+const getImageDimensions = function (imageDataUrl) {
+  return new Promise(resolve => {
+    let i = document.createElement('img');
+    i.onload = () => resolve({ width: i.width, height: i.height });
+    i.src = imageDataUrl;
+  });
+};
+
+class ImageValidationError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = 'ImageValidationError';
+  }
+}
+
+export default {
+  props: {
+    assessment: Object,
+    isEditing: Boolean,
+    errors: Array,
+    fileInputTypes: { type: Array, default: () => ['image/jpeg', 'image/png'] }
+  },
+  data() {
+    return {
+      fieldErrors: []
+    };
+  },
+  computed: {
+    answers() {
+      return this.assessment.answers;
+    },
+    correct() {
+      return this.assessment.correct;
+    },
+    feedback() {
+      return this.assessment.feedback;
+    },
+    disabled() {
+      return !this.isEditing;
+    },
+    imageInputType() {
+      return this.fileInputTypes.join(',');
+    }
+  },
+  methods: {
+    update(data) {
+      this.$emit('update', data);
+    },
+    toggleAnswer(index) {
+      let correct = cloneDeep(this.correct);
+      const position = correct.indexOf(index);
+
+      if (position < 0) {
+        correct.push(index);
+      } else {
+        correct.splice(position, 1);
+      }
+
+      this.update({ correct });
+    },
+    updateAnswer(evt) {
+      let answers = cloneDeep(this.answers);
+      const { target: input } = evt;
+      const { dataset, files = [] } = input;
+      const imageFile = files[0];
+
+      return new Promise((resolve, reject) => {
+        this.resetErrorMessages(dataset.index);
+        if (imageFile.size > 100000) {
+          return reject(new ImageValidationError(
+            'The image is too large. Maximum allowed image size: 100 kB'
+          ));
+        }
+        return resolve(imageFile);
+      })
+        .then(() => blobToDataURL(imageFile))
+        .then(imageDataUrl => {
+          answers[dataset.index] = imageDataUrl;
+          return getImageDimensions(imageDataUrl);
+        })
+        .then(({ width, height }) => {
+          if (width > maxImageDimension || height > maxImageDimension) {
+            return Promise.reject(new ImageValidationError(
+              `Incorrect image width and/or height. Maximum allowed image dimensions:
+              ${maxImageDimension}x${maxImageDimension}`
+            ));
+          }
+        })
+        .catch(error => {
+          answers[dataset.index] = '';
+          if (error.name === 'ImageValidationError') {
+            this.addErrorMessage(dataset.index, error.message);
+          } else {
+            throw error;
+          }
```

Comment:
As I said earlier on, use `instance of` instead and combine it with early return.